extensions/control-plane/provision/provision-gcs/src/main/java/org/eclipse/edc/connector/provision/gcp/GcsProvisioner.java

```diff
@@ -89,28 +103,23 @@ public CompletableFuture<StatusResult<ProvisionResponse>> provision(
         }
     }
 
+    private String getServiceAccountName(GcsResourceDefinition resourceDefinition) {
+        if (resourceDefinition.getServiceAccountName() != null) {
+            // TODO verify service account name from resource definition before returning.
+            return resourceDefinition.getServiceAccountName();
+        }
+
+        return gcpConfiguration.getServiceAccountName();
+    }
+
     @Override
     public CompletableFuture<StatusResult<DeprovisionedResource>> deprovision(
             GcsProvisionedResource provisionedResource, Policy policy) {
-        try {
-            iamService.deleteServiceAccountIfExists(
-                    new GcpServiceAccount(provisionedResource.getServiceAccountEmail(),
-                            provisionedResource.getServiceAccountName(), ""));
-        } catch (GcpException e) {
-            return completedFuture(StatusResult.failure(ResponseStatus.FATAL_ERROR,
-                    String.format("Deprovision failed with: %s", e.getMessage())));
-        }
         return CompletableFuture.completedFuture(StatusResult.success(
                 DeprovisionedResource.Builder.newInstance()
                         .provisionedResourceId(provisionedResource.getId()).build()));
     }
 
-    private GcpServiceAccount createServiceAccount(String processId, String buckedName) {
-        var serviceAccountName = sanitizeServiceAccountName(processId);
-        var uniqueServiceAccountDescription = generateUniqueServiceAccountDescription(processId, buckedName);
-        return iamService.getOrCreateServiceAccount(serviceAccountName, uniqueServiceAccountDescription);
-    }
-
     @NotNull
     private String sanitizeServiceAccountName(String processId) {
```

Comment:
unused (same for other 2 methods in this class)

Reply:
Removed